### ISSUE TITLE
It takes 55 seconds to detect a dog in doge.jpg

### ISSUE THREAD

**Issue by whatohyou** [matterport/Mask_RCNN#1895]
I used,
Mask_RCNN/samples/demo.ipynb

# Load a random image from the images folder
#file_names = next(os.walk(IMAGE_DIR))[2]
filename = os.path.join(IMAGE_DIR, 'doge.jpg')
#image = skimage.io.imread(os.path.join(IMAGE_DIR, random.choice(file_names)))
image= skimage.io.imread(filename)
#image= resize(image, (256, 256))
print('Start detecting~!')
start = time.time() # What in other posts is described is
# Run detection
results = model.detect([image], verbose=1)
end = time.time()         
stopwatch1 =end-start #Use then my code
print('Done!!! sec : '+ str(stopwatch1))

# Visualize results
r = results[0]
visualize.display_instances(image, r['rois'], r['masks'], r['class_ids'], class_names, r['scores'])

Start detecting~!
Processing 1 images
image                    shape: (480, 640, 3)         min:    0.00000  max:  255.00000  uint8
molded_images            shape: (1, 1024, 1024, 3)    min: -123.70000  max:  123.30000  float64
image_metas              shape: (1, 93)               min:    0.00000  max: 1024.00000  float64
anchors                  shape: (1, 65472, 4)         min:   -0.35390  max:    1.22878  float32
Done!!! sec : 48.70578598976135

![result_doge](https://user-images.githubusercontent.com/27909719/70365205-3694d880-18f5-11ea-83fa-90f44cd52400.png)

**Comment by whatohyou** [comment]
Is it normal to take so long to detect? Without gpu support it takes 75 seconds to detect the same  doge.jpg. When I look at gpu usage, gpu only works instantly at the last moment of detection. So when model.detect command was issued, it works on something for about 40+ seconds and gpu usage is 0% during that time. Then at the last moment gpu usage becomes 99% for like 0.2 seconds then back to 0% then result picture is displayed.
Anybody else have to same problem?

**Comment by marcfielding1** [comment]
Err so from your description I'm guessing the GPU goes nuts at the last second to render the result, could you show your inference log and requirements.txt please? You're using tensorflow-gpu I take it?

**Comment by whatohyou** [comment]
I upgraded windows 7 to windows 10 then reinstalled everything.
Time spent to detect is now 7.47773003578186 seconds.
Where can I find inference log?

requirement-gpu.txt
cython
scikit-image
setuptools>=41.0.0
progressbar2==3.46.1
pip>=19.0.0
absl-py==0.7.1
astor==0.8.0
attrs==19.1.0
backcall==0.1.0
bleach==3.1.0
certifi==2019.6.16
chardet==3.0.4
cycler==0.10.0
decorator==4.4.0
defusedxml==0.6.0
entrypoints==0.3
gast==0.2.2
google-pasta==0.1.7
grpcio==1.22.0
h5py==2.9.0
idna==2.8
ipykernel==5.1.1
ipython==7.6.1
ipython-genutils==0.2.0
ipywidgets==7.5.0
jedi==0.14.0
Jinja2==2.10.1
joblib==0.13.2
jsonschema==3.0.1
jupyter==1.0.0
jupyter-client==5.3.0
jupyter-console==6.0.0
jupyter-core==4.5.0
Keras==2.2.4
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
kiwisolver==1.1.0
Markdown==3.1.1
MarkupSafe==1.1.1
matplotlib==3.0.3
mistune==0.8.4
mpmath==1.1.0
nbconvert==5.5.0
nbformat==4.4.0
notebook==5.7.8
numpy==1.16.4
opencv-python==4.1.0.25
pandas==0.24.2
pandocfilters==1.4.2
parso==0.5.0
pexpect==4.7.0
pickleshare==0.7.5
Pillow==6.2.1
prometheus-client==0.7.1
prompt-toolkit==2.0.9
protobuf==3.8.0
ptyprocess==0.6.0
Pygments==2.4.2
pyparsing==2.4.0
pyrsistent==0.15.3
python-dateutil==2.8.0
pytz==2019.1
PyYAML==5.1.1
pyzmq==18.0.2
qtconsole==4.5.1
requests==2.22.0
scikit-learn==0.21.2
scipy==1.3.0
Send2Trash==1.5.0
six==1.12.0
sklearn==0.0
sympy==1.4
tensorboard==1.14.0
tensorflow-gpu==1.14.0
tensorflow-estimator==1.14.0
termcolor==1.1.0
terminado==0.8.2
testpath==0.4.2
tornado==6.0.3
traitlets==4.3.2
urllib3==1.25.3
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.15.4
widgetsnbextension==3.5.0
wrapt==1.11.2

**Comment by rakehsaleem** [comment]
It's taking a lot of time, sorry no idea about the reason but normally it should take less time.
An average 4k image (2268x4032) takes 1.8 seconds with CPU support with 16GB RAM.

Could you tell us the information for inference? 
You can find inference log in the second cell of balloon example.
==> Mask_RCNN/samples/demo.ipynb
```
 class InferenceConfig(crack.CrackConfig):
    # Set batch size to 1 since we'll be running inference on
    # one image at a time. Batch size = GPU_COUNT * IMAGES_PER_GPU
    GPU_COUNT = 1
    IMAGES_PER_GPU = 1

config = InferenceConfig()
config.display()
```

Hope it helps,
Thank you
